src/test/java/ir/sahab/walconsumer/WalConsumerTest.java

```diff
@@ -80,6 +94,52 @@ public void testWalConsumer() throws Exception {
         }
     }
 
+    /**
+     * In this test, we use more than one instances of wal consumers concurrently. According to our design, it is
+     * guaranteed that each item will be processed just once, and they will be processed sequentially (i.e., in any
+     * point in time there is just one consumer who is processing the head item from WAL).
+     */
+    @Test
+    public void testWalConsumingMultipleThreads() throws Exception {
+        final int numConsumers = 10;
+        final double numEntities = 100;
+
+        // Start WAL consumers.
+        WalConsumer[] consumers = new WalConsumer[numConsumers];
+        for (int i = 0; i < numConsumers; i++) {
+            consumers[i] = new WalConsumer(TestWalEntity.class, this::synchronizeModel, sessionFactory, "my_app");
```

Comment:
We can not check the order using HashMap, so for this test, it is better to use a Vector or CopyOnWriteArrayList (which are thread-safe) and check the exact order on them. So please define two methods: synchronizeMap (for the first test) and synchronizeVector (for the second test). Also rename the data structures to targetMap and targetVector.